saleor/attribute/models/base.py

```diff
@@ -242,23 +242,27 @@ def bulk_get_or_create(self, objects_data):
         results = []
         objects_not_in_db: List[AttributeValue] = []
 
+        # prepare a list that will save order index of attribute values
+        objects_enumerated = [
+            (index, obj_data) for index, obj_data in enumerate(objects_data)
+        ]
         query = self._prepare_query_for_bulk_operation(objects_data)
 
         # iterate over all records in db and check if they match any of objects data
         for record in query.iterator():
             # iterate over all objects data and check if they match any of records in db
-            for obj in objects_data:
+            for index, obj in objects_enumerated:
                 if self._is_correct_record(record, obj):
                     # upon finding existing record add it to results
-                    results.append(record)
-
-                    # remove it from objects data, so it won't be added to new records
-                    objects_data.remove(obj)
-
-                    break
+                    results.append((index, record))
+                    # remove it from objects list, so it won't be added to new records
+                    objects_enumerated.remove((index, obj))
```

Comment:
I don't think that removing the object from the list while iterating is correct approach.
```
In [17]: aa = ["a", "b", "c", "d", "e", "f"]

In [18]: enumerated_aa = list(enumerate(aa))

In [19]: print(enumerated_aa)
[(0, 'a'), (1, 'b'), (2, 'c'), (3, 'd'), (4, 'e'), (5, 'f')]

In [20]: for index, ob in enumerated_aa:
    ...:     print(index, ob)
    ...:     if index==1:
    ...:         print("removing ", ob)
    ...:         enumerated_aa.remove((index, ob))
    ...:
0 a
1 b
removing  b
3 d
4 e
5 f
```
See that the one iteration call is missing.

Reply:
This case here is different, whole goal of removing record from `objects_enumerated` is to reduce calls in the next iterations when looking for correct record

Reply:
But according to what @korycins sent, there is a chance that you will skip the validation for the record, when the record before this record will be removed

Reply:
At https://github.com/saleor/saleor/pull/15269/files#diff-14660fe9dd560e3979262f1c821c635bd3d367592a9f85822a1237235b606249R259 there is a `break` that will stop iteration and move to next db record and validation won't be skipped.